src/components/chart/statistics-charts.ts

```diff
@@ -0,0 +1,305 @@
+
+  @property({ attribute: false }) public statisticsData!: Statistics;
+
+  @property() public names: boolean | Record<string, string> = false;
+
+  @property({ attribute: false }) public endTime?: Date;
+
+  @property({ type: Array }) public statTypes: Array<StatisticType> = [
+    "sum",
+    "min",
+    "max",
+    "mean",
+  ];
+
+  @property({ type: Boolean, attribute: "up-to-now" }) public upToNow = false;
+
+  @property({ type: Boolean, attribute: "no-single" }) public noSingle = false;
+
+  @property({ type: Boolean }) public isLoadingData = false;
+
+  @state() private _chartData?: ChartData<"line">;
+
+  @state() private _chartOptions?: ChartOptions<"line">;
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return !(changedProps.size === 1 && changedProps.has("hass"));
+  }
+
+  public willUpdate(changedProps: PropertyValues) {
+    if (!this.hasUpdated) {
+      this._chartOptions = {
+        parsing: false,
+        animation: false,
+        scales: {
+          x: {
+            type: "time",
+            adapters: {
+              date: {
+                locale: this.hass.locale,
+              },
+            },
+            ticks: {
+              maxRotation: 0,
+              sampleSize: 5,
+              autoSkipPadding: 20,
+              major: {
+                enabled: true,
+              },
+              font: (context) =>
+                context.tick && context.tick.major
+                  ? ({ weight: "bold" } as any)
+                  : {},
+            },
+            time: {
+              tooltipFormat: "datetimeseconds",
+            },
+          },
+          y: {
+            ticks: {
+              maxTicksLimit: 7,
+            },
+          },
+        },
+        plugins: {
+          tooltip: {
+            mode: "nearest",
+            callbacks: {
+              label: (context) =>
+                `${context.dataset.label}: ${context.parsed.y}`,
+            },
+          },
+          filler: {
+            propagate: true,
+          },
+          legend: {
+            display: false,
+            labels: {
+              usePointStyle: true,
+            },
+          },
+        },
+        hover: {
+          mode: "nearest",
+        },
+        elements: {
+          line: {
+            tension: 0.4,
+            borderWidth: 1.5,
+          },
+          point: {
+            hitRadius: 5,
+          },
+        },
+      };
+    }
+    if (changedProps.has("statisticsData")) {
+      this._generateData();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!isComponentLoaded(this.hass, "history")) {
+      return html`<div class="info">
+        ${this.hass.localize("ui.components.history_charts.history_disabled")}
+      </div>`;
+    }
+
+    if (this.isLoadingData && !this.statisticsData) {
+      return html`<div class="info">
+        ${this.hass.localize(
+          "ui.components.statistics_charts.loading_statistics"
```

Comment:
We can share these strings with the history-based chart. It's all data. The user doesn't have to know there is a difference between both.

Reply:
It is all data, but statistics can also be future and not just history ;-)